src/target/semihosting.c

```diff
@@ -723,7 +723,7 @@ int32_t semihosting_exit(target_s *const target, const semihosting_exit_reason_e
 	if (reason == EXIT_REASON_APPLICATION_EXIT)
 		tc_printf(target, "exit(%" PRIu32 ")\n", status_code);
 	else
-		tc_printf(target, "Exception trapped: %" PRIx32 " (%" PRIu32 ")\n", reason, status_code);
+		tc_printf(target, "Exception trapped: %" PRIx16 " (%" PRIu32 ")\n", reason, status_code);
```

Comment:
Here, due too how enums work in C, the change should probably be to `%x` as enumerations have an underling type of `int` (but are treated by the compiler as unsigned for things like formatting). We can't assume that `PRIx16` is set up to be wide enough for the numerical representation of the enumeration otherwise.